python/lsst/dax/apdb/cassandra/apdbCassandra.py

```diff
@@ -725,7 +725,36 @@ def getDiaForcedSources(
 
     def containsVisitDetector(self, visit: int, detector: int) -> bool:
         # docstring is inherited from a base class
-        raise NotImplementedError()
+        # The order of checks corresponds to order in store(), on potential
+        # store failure earlier tables have higher probability containing
+        # stored records. With per-partition tables there will be many tables
+        # in the list, but it is unlikely that we'll use that setup in
+        # production.
+        existing_tables = self._schema.existing_tables(ApdbTables.DiaSource, ApdbTables.DiaForcedSource)
+        tables_to_check = existing_tables[ApdbTables.DiaSource][:]
+        if self.config.use_insert_id:
+            tables_to_check.append(self._schema.tableName(ExtraTables.DiaSourceChunks))
+        tables_to_check.extend(existing_tables[ApdbTables.DiaForcedSource])
```

Comment:
`existing_tables` is declared below to be a `dict[ApdbTables, list[str]]`, so then `existing_tables[ApdbTables.DiaSource]` should be a `list[str]`. What do you need the `[:]` index for (especially since line 737 doesn't have one)?

Reply:
I want to make a copy of the `existing_tables[ApdbTables.DiaSource]` so that I do not modify the original list in place when I do `append/extend` below.